src/api/ApiCall.hh

```diff
@@ -56,7 +97,7 @@ struct ApiResponseBase
     /// <returns><c>true</c> if the call was unsuccessful, <c>false</c> if not.</returns>
     bool Failed() const noexcept
     {
-        return (Result == ApiResult::Error || Result == ApiResult::Unsupported || Result == ApiResult::Failed);
+        return (!Succeeded() && Result != ApiResult::None);
```

Comment:
Up to you but I think you should make the order of evaluation clearer by using parentheses.
Kind of like this: `!(Succeeded() && (Result == ApiResult::None))`
or: `(!Succeeded() && (Result != ApiResult::None))`

Reply:
Alright